### DISCUSSION TITLE
Build and transmit data without prior recording

### DISCUSSION THREAD

**Discussion by tooboot** [jopohl/urh#876] [General]
Hi,
Thank you for developing such a capable tool in URH. 
I'm sure to be missing something obvious but: how can I use the Generator or Simulator functions to simply type in some data (representing packet data), configure the modulation scheme and SDR frequency, and simply command URH to transmit that data? But, without having recorded something from an RF source first? The YouTube and user guide examples seem to show a workflow that requires a signal to be recorded before any decoded/manipulated/encoded parts of it can be transmitted again. But I don't need to import any IQ file data, or record a signal (because there is no signal to record yet) but to simply type in some bit or byte values and say "Tx now, with these RF specifications". What am I doing wrong?

**Comment by jopohl** [comment]
If you right click e.g. on the table in Generator you have the option to add a new empty message, afterwards you can follow the workflow you already saw.

**Reply by tooboot** [reply]
Perfect - thank you for your speedy reply.
